index.js

```diff
@@ -26,17 +26,26 @@ function getResources(dir, config) {
 /**
  * Smokes the given resources.
  */
-function smoke(resources, currentTry, config) {
+function smoke(resources, currentTry, config, callback) {
   shisha.smoke(resources, function(report){
-    var failedResources = _.filter(report, function(resource){
+    var failedResources = _(report).filter(function(resource){
       return !(resource.result);
-    })
+    }).map(function(item) {
+      item.url = item.url();
+      item.status = 200;
+      return item;
+    }).value();
 
     if (Object.keys(failedResources).length) {
       setTimeout(function(){
-        retry(failedResources, currentTry, config);
+        try {
+          retry(failedResources, currentTry, config, callback);
+        } catch (err) {
+          callback(err);
+        }
```

Comment:
I did it to have control on the error handling

When we are running we have 2 places where we run a function and (without callback) we lose control of what is happening:

```javascript
getResources(dir, config).then(function(resources){
  smoke(resources, currentTry, config); // we don't return so we don't wait
}).done()
```

and 

```javascript
setTimeout(function(){ // we execute this at some point in the future
  retry(failedResources, currentTry, config);
}, config.sleep * 1000)
```

With the callback we can decide what to do with the error. As of today we re-throw it because we want to stop the execution :)